### PR TITLE
Add developer documentation for the new editor

### PR DESCRIPTION
**Pull request by bwindels**
which I dubbed CIDER, for better or worse.

### CODE REVIEW

**Comment by jryans** [comment]
PR titles go into our change log, so it's better to use standard casing and text that clearly explains the change to the casual observer. I'll try tweaking this one, but keep it in mind for the future. 😄

File: docs/ciderEditor.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# The CIDER (Contenteditable-Input-Diff-Error-Reconcile) editor
+
+The CIDER editor is a custom editor written for Riot. Most of the code can be found in the `/editor/` directory of the `matrix-react-sdk` project. It is used to power the composer to edit messages, and will soon be used as the main composer to send messages as well.
```
Comment (jryans) [inline docs/ciderEditor.md]: Haha, I like the name, especially as a fan of cider myself... 😁

File: docs/ciderEditor.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# The CIDER (Contenteditable-Input-Diff-Error-Reconcile) editor
```
Comment (jryans) [inline docs/ciderEditor.md]: Hmm, I feel like docs should be wrapped to the same line length as code for easy reading in any editor, but we seem to be a bit all over the place with doc line lengths... 😭

File: docs/ciderEditor.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+The CIDER editor is a custom editor written for Riot. Most of the code can be found in the `/editor/` directory of the `matrix-react-sdk` project. It is used to power the composer to edit messages, and will soon be used as the main composer to send messages as well.
+
+## High-level overview.
+
+The editor is backed by a model that contains parts. A part has some text and a type (plain text, pill, ...). When typing in the editor, the model validates the input and updates the parts. The parts are then reconciled with the DOM.
+
+## Inner workings
+
+When typing in the `contenteditable` element, the `input` event fires and the DOM of the editor is turned into a string. The way this is done has some logic to it to deal with adding newlines for block elements, ... so doesn't use `innerText`, `textContent` or anything similar. The model addresses any content in the editor within as an offset within this string. The caret position is thus also converted from a position in the DOM tree to an offset in the content string. This happens in `getCaretOffsetAndText` in `dom.js`.
```
Comment (jryans) [inline docs/ciderEditor.md]: What's the "..." here?

**Comment by bwindels** [comment]
Addressed the comments, so merging.

**Pull request merged**